mlflow/utils/model_utils.py

```diff
@@ -149,3 +149,45 @@ def _add_code_from_conf_to_system_path(local_path, conf, code_key=FLAVOR_CONFIG_
     if code_key in conf and conf[code_key]:
         code_path = os.path.join(local_path, conf[code_key])
         _add_code_to_system_path(code_path)
+
+
+def _validate_onnx_session_options(onnx_session_options):
+    """
+    Validates that the specified onnx_session_options dict is valid.
+
+    :param ort_session_options: The onnx_session_options dict to validate.
+    """
+    import onnxruntime as ort
+
+    if onnx_session_options is not None:
+        if not isinstance(onnx_session_options, dict):
+            raise TypeError(
+                "Argument onnx_session_options should be a dict, not {}".format(
+                    type(onnx_session_options)
+                )
+            )
+        for key, value in onnx_session_options.items():
+            if key != "extra_session_config" and not hasattr(ort.SessionOptions, key):
+                raise ValueError(
+                    f"Key {key} in onnx_session_options is not a valid \
+                        ONNX Runtime session options key"
+                )
```

Comment:
This is how the actual error message look like:

```
key = 1
raise ValueError(
    f"Key {key} in onnx_session_options is not a valid \
                        ONNX Runtime session options key"
)
```

```
Traceback (most recent call last):
  File "a.py", line 2, in <module>
    raise ValueError(
ValueError: Key 1 in onnx_session_options is not a valid                         ONNX Runtime session options key
```

Reply:
Good catch!